Create 1.31 release notes (#1012)
okteto/docs: src/content/release-notes.mdx

id: release-notes
---

## 1.31.0

4 April 2025

This version is compatible with Kubernetes versions 1.30 to 1.32 \
Okteto Chart release 1.31 is designed to work with [Okteto CLI 3.6.x](https://github.com/okteto/okteto/releases/tag/3.6.0)

### Deprecation Notice {#deprecation-notice-1.31}

> - ⚠️ Important: **Support for Docker Image Manifest Schema 1 images is removed in this** (1.32) due to upstream dependency changes. If you are using older images, they may fail to pull or deploy.\

Reviewer: ```suggestion
- ⚠️ Important: **Support for Docker Image Manifest Schema 1 images is removed in this** (1.31) due to upstream dependency changes. If you are using older images, they may fail to pull or deploy.\
```

We refer to this version we are releasing, right?

[Learn how to check and update your images →](self-hosted/manage/upgrade.mdx#upgrading-to-okteto-131x)

### New Features {#new-features-1.31}

- Added a new [Okteto API endpoint](admin/okteto-api.mdx) to list users  <!-- 9175 -->

> - Added support for waking up Development Environments via Docker Compose annotations; `depends_on` is now respected [with a feature flag](reference/feature-flags.mdx)  <!-- 9187 -->

Reviewer: ```suggestion
- Added support for waking up resources within Development Environments respecting the dependencies defined through `depends_on` field in Docker Compose. By default this behavior is disabled, but it can be enabled [with the feature flag](reference/feature-flags.mdx) `OKTETO_COMPOSE_WAIT_FOR_DEPENDENCIES`  <!-- 9187 -->
```

Just a suggestion for a couple of things:
* The first sentence was giving me the impression that we could wake up a namespace using Docker Compose annotations. Which it wasn't the case. We just use annotations to be able to know the order when waking up the resources
* I think it is easier for the users if we mention the feature flag specifically in the release notes, so they don't need to look through all the feature flags to locate the one they are interested on. I think it is important to keep the link to the feature flags page for discoverability, and to get familiar with them if they never used them, but we should also include the specific flag name

- Added support for Kubernetes [1.32](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.32.md) (previous support for 1.29 was removed)

### Improvements {#improvements-1.31}

> - Improved error messages when ssh-keyscan error fails  <!-- https://github.com/okteto/app/pull/9197 -->

Reviewer: ```suggestion
- Improved error messages when ssh-keyscan error fails for deploys triggered from the UI, `okteto pipeline deploy` or `okteto preview deploy`  <!-- https://github.com/okteto/app/pull/9197 -->
```

Just to give some more context to users to indicate when this could happen. This is something internal of Okteto on specific operations

id: release-notes
---

## 1.31.0

4 April 2025

> This version is compatible with Kubernetes versions 1.29 to 1.32 \

Reviewer 2: 1.29 is not supported any more based on our new kubernetes policy

Okteto Chart release 1.31 is designed to work with [Okteto CLI 3.6.x](https://github.com/okteto/okteto/releases/tag/3.6.0)

### Deprecation Notice {#deprecation-notice-1.31}

- ⚠️ Important: **Support for Docker Image Manifest Schema 1 images is removed in this** (1.32) due to upstream dependency changes. If you are using older images, they may fail to pull or deploy.\
[Learn how to check and update your images →](self-hosted/manage/upgrade.mdx#upgrading-to-okteto-131x)

### New Features {#new-features-1.31}

- Added a new Public API endpoint to list users  <!-- 9175 -->

> - Added support for waking up Development Environments based on annotations in Docker Compose files  <!-- 9187 -->

Reviewer: This is the same than the first point in Improvements. This refers to the fact of respecting the `depends_on` order for compose deployments. So we should keep one or the other

> - Optimized the wake-up process to allow resources and Development Environments to start in parallel

Reviewer: This won't be included in the release. It is still on review, so it didn't make it

- Added support for Kubernetes 1.32 (previous support for 1.29 was removed)

### Improvements {#improvements-1.31}

> - Namespace wake-up process will now respect Docker Compose `depends_on` with a feature flag

Reviewer: @jLopezbarb Could you send a PR including the feature flag we added to enabled the behavior?

@codyjlandstrom Should we reference here the name of the feature flag, so it is easily recognized by our users?

- Improved error messages when ssh-keyscan error fails  <!-- https://github.com/okteto/app/pull/9197 -->
- Patched IngressNightmare CVE-2025-1974 to enhance platform security  <!-- https://github.com/okteto/app/pull/9205 -->
- Upgraded our BuildKit client to [0.20.2](https://github.com/moby/buildkit/releases/tag/v0.20.2)

### Bug Fixes {#bug-fixes-1.31}

> - The `baggage.divert` header is now properly propagated to downstream services to indicate when a request is being diverted

Reviewer 2: This is specific for the `nginx` divert driver

- Fixed an issue where logs appeared out of order on initial load  <!-- https://github.com/okteto/app/pull/9139 -->

> - Support bundles now include Ingress NGINX logs and Helm values for installations managed via ArgoCD

Reviewer: I'm not sure if this is interesting for our customers (I mean, not sure if it is relevant for them to include it in the release notes)

Author: Going to leave it for now as we did add it based on a customer conversation

> - Updated Helm chart to include Okteto CLI 3.5.1, which resolves issues with Smart Builds  <!-- https://github.com/okteto/app/pull/9209 -->

Reviewer: This doesn't apply as this version includes CLI `3.6.x`, which means that includes the Smart Builds fix. As this was already solve on a previous version, we don't include it in the next onex
